.github/workflows/cmake.yml

```diff
@@ -0,0 +1,90 @@
+# inspired by "build GDAL library and to run test with CMake"
+# from https://github.com/OSGeo/gdal/blob/master/.github/workflows/cmake_builds.yml
+
+name: Build and test with CMake
+
+on:
+    push:
+        paths-ignore:
+            - 'doc/**'
+        branches-ignore:
+            - 'backport**'
+    pull_request:
+        paths-ignore:
+            - 'doc/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+env:
+  CMAKE_UNITY_BUILD: OFF
+
+permissions:
+  contents: read  #  to fetch code (actions/checkout)
+
+jobs:
+
+  build-linux-ubuntu-focal:
+    runs-on: ubuntu-20.04
+    env:
+      CMAKE_OPTIONS: -DPython_LOOKUP_VERSION=3.8 -DUSE_CCACHE=ON
+      cache-name: cmake-ubuntu-focal
+    steps:
+    - name: Checkout GRASS
+      uses: actions/checkout@v3
+    - name: Setup cache
+      uses: actions/cache@v3
+      id: cache
+      with:
+          path: ${{ github.workspace }}/.ccache
+          key: ${{ runner.os }}-${{ env.cache-name }}-${{ github.base_ref }}${{ github.ref_name }}-${{ github.run_id }}
+          restore-keys: |
+            ${{ runner.os }}-${{ env.cache-name }}-${{ github.base_ref }}
+            ${{ runner.os }}-${{ env.cache-name }}
+    - name: Install CMake 3.9.1
```

Comment:
Is there a specific reason that cmake is installed that way on the runner? I understand that it isn't really for the GitHub runners, since cmake is installed by default (un ubuntu-latest (22.04LTS) it is currently cmake 3.25.1), but for documenting what was used, maybe for local porting. However, GitHub workflow files are already not runnable as is locally (I'm not talking about self-hosted runners here).
If there was no specific reason for using that version of cmake (released August 10, 2017), we could take it out, install one from  apt-get (ubuntu), or use a GitHub action step that is integrated, like https://github.com/marketplace/actions/get-cmake#if-you-want-to-use--latest-stable-you-can-use-this-one-liner

Reply:
> Is there a specific reason that cmake is installed that way on the runner?

This was only my guess-work, maybe totally wrong. Here I took inspiration from the GDAL workflow while having zero knowledge about cmake :-)